### PR TITLE
dotnet 6.0 update to Hub pages

### PR DESCRIPTION
**Pull request by wadepickett**
[Internal Review](https://review.docs.microsoft.com/en-us/dotnet/?branch=pr-en-us-26869)

First round of immediate updates to align with .NET 6.0:
Contributes to #21005

- Updated links for What's New, reference and any other links where a .NET 6.0 specific topic is now available and the version has to be specified to get there.
- Reviewed past data for this hub page to see what might need a change.
- Reduced the number of top highlight links to reflect top most used links, and where top traffic is to be driven to.
- Removed original Maui links, new links will be added as owners provide new ones in a future PR.

### CODE REVIEW

**Comment by wadepickett** [comment]
@BillWagner & @gewarren,   These are just the immediate changes that need to occur for just the Hub page.  Landing page updates will be a separate PR.

File: docs/index.yml [hunk 1]
```diff
@@ -298,7 +290,7 @@ additionalContent:
       # Card
       - title: ".NET API reference"
         summary: API reference documentation for .NET
-        url: ../api/index.md?view=net-5.0
+        url: ../api/index.md?view=net-6.0
       # Card
       - title: ".NET Core API reference"
```
Comment (gewarren) [inline docs/index.yml:295]: I don't think we should have separate cards for .NET and .NET Core API ref. So I suggest removing the .NET Core API ref card.

**Review by gewarren — APPROVED** [review]
Otherwise, looks great.

**Comment by wadepickett** [comment]
Thanks for the review.  Just waiting for a build at this point, builds are slow today...

**Review by gewarren — APPROVED** [review]

**Pull request merged**